File: app/src/main/java/com/moudle/app/view/RoundDrawable.java
package com.moudle.app.view;

import android.graphics.Bitmap;
import android.graphics.BitmapShader;
import android.graphics.Canvas;
import android.graphics.ColorFilter;
import android.graphics.Paint;
import android.graphics.PixelFormat;
import android.graphics.Shader;
import android.graphics.drawable.Drawable;

import androidx.annotation.IntRange;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * 自定义圆形图片
 */
public class RoundDrawable extends Drawable {

    private final Paint mPaint;
    private final int mWidth;

    public RoundDrawable(Bitmap bitmap) {

        BitmapShader bitmapShader = new BitmapShader(bitmap, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP);
        mPaint = new Paint();
        mPaint.setAntiAlias(true);
        mPaint.setShader(bitmapShader);

        mWidth = Math.min(bitmap.getWidth(), bitmap.getHeight());
    }

    @Override
    public void draw(@NonNull Canvas canvas) {
        canvas.drawCircle(mWidth >> 1, mWidth >> 2, mWidth >> 2, mPaint);
    }

    @Override
    public void setAlpha(@IntRange(from = 0, to = 255) int i) {
        mPaint.setAlpha(i);
    }

    @Override
    public void setColorFilter(@Nullable ColorFilter colorFilter) {
        mPaint.setColorFilter(colorFilter);
    }

    @Override
    public int getOpacity() {
        return PixelFormat.TRANSLUCENT;
    }

    @Override
    public int getIntrinsicHeight() {
        return mWidth;
    }

    @Override
    public int getIntrinsicWidth() {
        return mWidth;
    }
}

File: app/src/main/java/com/moudle/app/view/RoundDrawable.kt
package com.moudle.app.view

import android.graphics.*
import android.graphics.drawable.Drawable
import androidx.annotation.IntRange

/**
 * @作者: Simplation
 * @日期: 2021/6/29 10:45
 * @描述:
 * @更新:
 */
class RoundDrawable(bitmap: Bitmap) : Drawable() {
    private val mPaint: Paint
    private val mBitmap: Bitmap = bitmap
    private val mWidth: Int

    init {
        val bitmapShader = BitmapShader(mBitmap, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP)
        mPaint = Paint()
        mPaint.isAntiAlias = true
        mPaint.shader = bitmapShader
        mWidth = mBitmap.width.coerceAtMost(mBitmap.height)
    }

    override fun draw(canvas: Canvas) {
        canvas.drawCircle(
            (mWidth / 2).toFloat(),
            (mWidth / 2).toFloat(),
            (mWidth / 2).toFloat(),
            mPaint
        )
    }

    override fun setAlpha(@IntRange(from = 0, to = 255) i: Int) {
        mPaint.alpha = i
    }

    override fun setColorFilter(colorFilter: ColorFilter?) {
        mPaint.colorFilter = colorFilter
    }

    override fun getOpacity(): Int {
        return PixelFormat.TRANSLUCENT
    }

    override fun getIntrinsicHeight(): Int {
        return mWidth
    }

    override fun getIntrinsicWidth(): Int {
        return mWidth
    }
}